assets/user-profile/components/PauseNotificationModal.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import {connect} from 'react-redux';
+import {IUser} from 'interfaces';
+import {gettext} from 'utils';
+import {modalFormInvalid, modalFormValid} from 'actions';
+import {updateUserNotificationPause} from 'user-profile/actions';
+import Modal from 'components/Modal';
+
+interface IProps {
+    modalFormInvalid(): void;
+    modalFormValid(): void;
+    updateUserNotificationPause(schedule: Omit<IUser['notification_schedule'], 'last_run_time'>): void;
+    data: {
+        user: IUser;
+    };
+}
+
+interface IState {
+    pauseFrom?: string;
```

Comment:
if you are applying defaults in the constructor, should properties not be mandatory then?